java/org/apache/catalina/util/TLSUtil.java

```diff
@@ -38,6 +38,8 @@ public static boolean isTLSRequestAttribute(String name) {
                 Globals.KEY_SIZE_ATTR.equals(name)  ||
                 Globals.SSL_SESSION_ID_ATTR.equals(name) ||
                 Globals.SSL_SESSION_MGR_ATTR.equals(name) ||
-                SSLSupport.PROTOCOL_VERSION_KEY.equals(name);
+                SSLSupport.PROTOCOL_VERSION_KEY.equals(name) ||
+                SSLSupport.REQUESTED_PROTOCOL_VERSIONS_KEY.equals(name) ||
+                SSLSupport.REQUESTED_CIPHERS_KEY.equals(name);
```

Comment:
I think this would look and read nicer if we use `switch (String)`. Extra benefit: it will be faster!

Reply:
Will do.